app/services/user_seeder.rb

```diff
@@ -0,0 +1,122 @@
+class UserSeeder
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  def initialize(**options)
+    @csv_file = options[:csv_file]
+    file = File.read(csv_file)
+    @csv = CSV.parse(file, headers: true)
+    validate_csv_headers
+    @email_domain = options[:email_domain]
+    validate_email_domain
+    @deploy_env = options[:deploy_env] || LoginGov::Hostdata.env
+  rescue Errno::ENOENT
+    raise ArgumentError.new("#{csv_file} does not exist")
+  end
+
+  # Seed the users whose PII is specified in the CSV file. Raises an
+  # ArgumentError if an email address is already taken and uses a transaction to
+  # avoid partial completion.
+  #
+  # @return [Integer] the number of users created
+  def run
+    validate_environment
+
+    ActiveRecord::Base.transaction do
+      @csv.each_with_index do |row, i|
+        email = "user#{i}@#{email_domain}"
+        row['email_address'] = email
+        row['password'] = PASSWORD
+        ee = EncryptedAttribute.new_from_decrypted(email)
+
+        User.create!(email_fingerprint: ee.fingerprint) do |user|
+          codes = setup_user(user: user, ee: ee)
+          row['codes'] = codes.join('|')
```

Comment:
can we unify any of this with the `dev.rake` task? Or if this is urgent, maybe create a follow-up JIRA ticket to do that?

Reply:
Follow up ticket sounds good - it seems like we'd split out the `create a user` part from the `parse CSV and validate stuff` part.